CDCgov/prime-reportstream
# 6972 Universal Pipeline can send HL7 BATCH files

This PR adds batching HL7 messages for the Universal Pipeline

Test Steps:
1. Run ReportStream
2. Send more than one message in quick succession as the `ignore.ignore-full-elr` sender.
3. Wait until the batch function runs.
4. Verify that there is one file per message you sent in the Blobstore batch folder for the `co.full-elr-hl7` receiver.
5. Verify that there is ONLY one file in the Blobstore ready folder for the the `co.full-elr-hl7` receive.  Verify the contents of the file is an HL7 Batch file with all the messages you sent.

## Changes
- Added clearDB to gradle to just clear up all report related data without deleting settings.
- Separated UP batching code into its own function
- Unit tests specific to UP batching

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
